src/field/RungeKuttaStepper.hh

```diff
@@ -0,0 +1,83 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file RungeKuttaStepper.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Range.hh"
+#include "base/Types.hh"
+#include "base/Macros.hh"
+//#include "FieldEquation.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * The 4th order classical Runge-Kutta stepper
+ *
+ * This method estimates the updated state from an initial state and slope
+ * estimate, with fourth-order accuracy.
+ *
+ * For a magnetic field equation \em f with state \em x, which includes
+ * position and momentum but may also include time and spin:
+ * \f[
+ *  XXX TODO: document steps here in latex math
+ * \f]
+ */
+template<class FieldEquation_T>
+class RungeKuttaStepper
+{
+    //@{
+    //! Type aliases
+    using OdeState = typename FieldEquation_T::OdeState;
+    //@}
+
+  public:
+    //! Construct with the equation of motion
+    inline CELER_FUNCTION RungeKuttaStepper(FieldEquation_T& equation);
+
+    //! Adaptive step size control
+    CELER_FUNCTION auto operator()(real_type       step,
+                                   const OdeState& beg_state,
+                                   const OdeState& beg_slope) -> OdeState;
+
+    //! Stepper truncation error
+    CELER_FUNCTION real_type error(real_type step, const OdeState& beg_state);
+
+    // Closerest distance between the chord and the mid-point
+    CELER_FUNCTION real_type distance_chord(const OdeState& beg_state,
+                                            const OdeState& end_state) const;
+
+    //! The right hand side of the field equation
+    CELER_FUNCTION OdeState ode_rhs(const OdeState& beg_state) const
+    {
+        return equation_(beg_state);
+    }
```

Comment:
The stepper shouldn't need an extra function to evaluate the equation. Whoever instantiates the class should be  able to call the equation independently if needed -- this is extra functionality that's not really part of being a "stepper".

Reply:
Removed - left over from an earlier implementation.